File: isma-intg-api/src/main/java/ru/nstu/isma/intg/api/methods/StabilityIntgController.java
package ru.nstu.isma.intg.api.methods;

/**
 * @author Maria Nasyrova
 * @since 25.02.2015
 */
public abstract class StabilityIntgController extends IntgController {

    public abstract double predictNextStepSize(IntgPoint toPoint);

    protected abstract double getStabilityInterval();

    protected abstract double getMaxJacobiEigenValue(IntgPoint point);

}

File: isma-intg-api/src/main/kotlin/ru/nstu/isma/intg/api/methods/StabilityIntgController.kt
package ru.nstu.isma.intg.api.methods

abstract class StabilityIntgController: IntgController() {
    abstract fun predictNextStepSize(toPoint: IntgPoint): Double

    protected abstract fun getStabilityInterval(): Double

    protected abstract fun getMaxJacobiEigenValue(point: IntgPoint): Double
}